src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -185,6 +197,39 @@ internal static bool TryGetBestOverload(TexlBinding txb, CallNode node, DType[]
             return false;
         }
 
+        /// <summary>
+        /// Typecheck the invocation and infer the return type. Error on deferred(unknown) arg is not considered.
+        /// </summary>
+        internal static bool HandleCheckInvocationWithDeferred(TexlFunction maybeFunc, TexlBinding txb, TexlNode[] args, DType[] argTypes, out ErrorContainer checkInvocationErrors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            // This error container is used as temporary container so we can trap type mismatch kind of error for
+            // deferred (unknown) type args and validate all the errors were caused due to deferred(unknown) type.
+            checkInvocationErrors = new ErrorContainer();
+
+            var typeCheckSucceeded = maybeFunc.HandleCheckInvocation(txb, args, argTypes, checkInvocationErrors, out returnType, out nodeToCoercedTypeMap);
+
+            // If type check failed and errors were due to Unknown type node we would like to consider the typeChecking passed.
+            if (!typeCheckSucceeded && checkInvocationErrors.HasErrors() && checkInvocationErrors.GetErrors().All(error => txb.GetType(error.Node).IsDeferred))
+            {
+                // If one of the arg was unknown and that generated error (e.g. type mismatch)
+                // and return type could not be calculated and was error we assign it as unknown.
+                // and if return type was Table, we assign it to be table of unknown, so operation like In can work.
+                switch (returnType.Kind)
+                {
+                    case DKind.Error:
+                        returnType = DType.Deferred;
+                        break;
+                    case DKind.Table:
```

Comment:
is there any reason we'd care about `Record` results here?

Reply:
That is so operator like `In` can work.
e.g. 123 in Table(X) would only work if Table(X) resolves to single column deferred type table. Because then type of one column table can be accepted by number type.